### PR TITLE
feat: make all internal rate limits configurable

### PR DESCRIPTION
**Pull request by chriswk**
### What
This PR makes the rate limit for user creation and simple login (our password based login) configurable in the same way you can do metricsRateLimiting.

### Worth noting
In addition this PR adds a `rate_limit{endpoint, method}` prometheus gauge, which gets the data from the UnleashConfig.

### CODE REVIEW

File: src/lib/create-config.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ import {
     ICspDomainOptions,
     IClientCachingOption,
     IMetricsRateLimiting,
+    IRateLimiting,
```
Comment (thomasheartman) [inline src/lib/create-config.ts:22]: Irate limiting? 😄

File: src/lib/create-config.ts [hunk 2]
```diff
@@ -132,6 +133,23 @@ function loadMetricsRateLimitingConfig(
...
 }
 
+function loadRateLimitingConfig(options: IUnleashOptions): IRateLimiting {
+    const createUserMaxPerMinute = parseEnvVarNumber(
+        process.env.CREATE_USER_RATE_LIMIT_PER_MINUTE,
+        20,
+    );
+    const simpleLoginMaxPerMinute = parseEnvVarNumber(
+        process.env.SIMPLE_LOGIN_LIMIT_PER_MINUTE,
+        10,
```
Comment (thomasheartman) [inline src/lib/create-config.ts:143]: Curious as to why one is called "x_RATE_LIMIT_PER_MINUTE" and the other is just called "x_LIMIT_PER_MINUTE"? Not a big deal, but feels like they should have similar names to make it easier to remember.

File: src/lib/metrics.ts [hunk 3]
```diff
@@ -259,6 +265,42 @@ export default class MetricsMonitor {
                         .labels({ range: clientStat.range })
                         .set(clientStat.count),
                 );
+
+                rateLimits.reset();
```
Comment (thomasheartman) [inline src/lib/metrics.ts:269]: Any particular reason we need to reset it here? Will we have gotten any inputs before this point that we need to get rid of?

File: src/lib/metrics.ts [hunk 4]
```diff
@@ -259,6 +265,42 @@ export default class MetricsMonitor {
...
+                        config.metricsRateLimiting.frontendMetricsMaxPerMinute,
+                    );
+                rateLimits
+                    .labels({
+                        endpoint: '/api/frontend/register',
+                        method: 'POST',
+                    })
+                    .set(
+                        config.metricsRateLimiting.frontendRegisterMaxPerMinute,
+                    );
```
Comment (thomasheartman) [inline src/lib/metrics.ts:294]: Out of curiosity: what were all of these before? If these options are old, how were they applied? Did we do it somewhere else?

**Review by thomasheartman — APPROVED** [review]
This looks pretty good to me! Have a couple of questions and suggestions, but nothing blocking 😄

File: src/lib/metrics.ts (hunk 3 above)
Comment (chriswk) [reply src/lib/metrics.ts:269]: I'm still not sure, I just followed the patterns for the rest of our static metrics.

Comment (chriswk) [reply src/lib/metrics.ts:269]: In my mind since it's a gauge there shouldn't really be a need to reset it.

File: src/lib/create-config.ts (hunk 1 above)
Comment (chriswk) [reply src/lib/create-config.ts:22]: indeed, I was quite IRate while writing this :P

File: src/lib/metrics.ts (hunk 4 above)
Comment (chriswk) [reply src/lib/metrics.ts:294]: there were no limits for register and metrics earlier besides what's configured on the loadbalancer side.
For the user creation and login it was 10 and 20.

**Pull request merged**